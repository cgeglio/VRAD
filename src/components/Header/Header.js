import React from 'react';
import PropTypes from 'prop-types';
import './Header.scss';
import { Link } from 'react-router-dom';

export const Header = ({ user, logout, favoritesNumber }) => {
  return (
    <header>
      <img src={process.env.PUBLIC_URL + '/scoutGreen.png'} alt="Logo" />
      <h2>Welcome {user.name}!</h2>
      <div>
        <Link to="/favorites"><button type="button" className="favorites">Favorites ({`${favoritesNumber || 0}`})</button></Link>
        <Link to="/"><button type="button" className="logOut" onClick={() => logout()}>Log out</button></Link>
      </div>
    </header>
  );
};

Header.propTypes = {
  user: PropTypes.object.isRequired,
  logout: PropTypes.func.isRequired,
  favoritesNumber: PropTypes.number,
};
