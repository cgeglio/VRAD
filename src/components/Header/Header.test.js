import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './Header';

describe('Header', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const mockUser = {
      name: 'Jeff',
      email: 'jeff@turing.org',
      purpose: 'vacation',
    };
    const wrapper = shallow(<Header
      user={mockUser}
      logout={jest.fn()}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
