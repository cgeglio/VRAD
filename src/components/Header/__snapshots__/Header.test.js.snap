// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot with all data passed in correctly 1`] = `
<header>
  <img
    alt="Logo"
    src="/scoutGreen.png"
  />
  <h2>
    Welcome 
    Jeff
    !
  </h2>
  <div>
    <Link
      to="/favorites"
    >
      <button
        className="favorites"
        type="button"
      >
        Favorites (
        0
        )
      </button>
    </Link>
    <Link
      to="/"
    >
      <button
        className="logOut"
        onClick={[Function]}
        type="button"
      >
        Log out
      </button>
    </Link>
  </div>
</header>
`;
